pandas/io/parsers.py

```diff
@@ -945,16 +947,16 @@ def _clean_options(self, options, engine):
         delim_whitespace = options["delim_whitespace"]
 
         # C engine not supported yet
-        if engine == "c":
+        if engine == "c" or engine == "arrow":
             if options["skipfooter"] > 0:
-                fallback_reason = "the 'c' engine does not support skipfooter"
+                fallback_reason = f"the {engine} engine does not support skipfooter"
                 engine = "python"
 
         encoding = sys.getfilesystemencoding() or "utf-8"
         if sep is None and not delim_whitespace:
-            if engine == "c":
+            if engine == "c" or engine == "arrow":
```

Comment:
Hmmm...I feel like we may need to create a variable (at line 948) like this:

~~~python
not_python_engine = engine == "c" or engine == "arrow"
~~~